### PR TITLE
FLPATH-41: Execute main workflow asynchronously

### PR DESCRIPTION
**Pull request by masayag**
Since the duration of a workflow might be long, when a workflow is being executed, it will be submitted to an executor and a response will return the immediate "IN_PROGRESS" status with the execution-id to allow monitoring of the progress by the caller.

Within the same patch, the async-executor of the continuation service is being replaced in favor of a common executor for any async calls.

WorkFlow Async Execution (1)

### CODE REVIEW

**Comment by RichardW98** [comment]
is the workflow racing issue ressolved?

**Comment by masayag** [comment]
> is the workflow racing issue resolved?

Some of the examples were able to complete. Other requires to be updated due to the change in the nature of the call.
I haven't seen issues with the execution though.
So I don't suspect there is a race here. However, I'll switch from h2 to postgres so I can debug properly (in h2 the console froze when there is a breakpoint enabled...).

I was somewhat misled by other error that appears in the logs, but seems not to be related to my changes.

**Comment by masayag** [comment]
Depends on https://github.com/parodos-dev/parodos/pull/273

File: workflow-examples/run_examples.sh
[the remark this answers is not in the document]
Comment (masayag) [reply workflow-examples/run_examples.sh:1]: Yes, but that can be done only once https://github.com/parodos-dev/parodos/pull/273 is merged, so there will be an option to query for the assessment retrieved workflow options.

[the remark this answers is not in the document]
Comment (masayag) [reply workflow-examples/run_examples.sh:1]: Update the sdk based tests as well.

**Comment by masayag** [comment]
The PR should be merged together with UI PR to avoid compatibility issues.

**Comment by eloycoto** [comment]
I'm sorry for the trouble!

**Comment by masayag** [comment]
@dagda1 this PR should be merged in accordance with parodos-backstage PR.
Pls update when it can.

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java [hunk 1]
```diff
@@ -119,31 +124,26 @@ private void statusCounterWithStatus(WorkFlowStatus status) {
...
-				projectId);
-		if (executionId != null)
-			WorkContextDelegate.write(workContext, WorkContextDelegate.ProcessType.WORKFLOW_EXECUTION,
-					WorkContextDelegate.Resource.ID, executionId.toString());
-		WorkContextDelegate.write(workContext, WorkContextDelegate.ProcessType.WORKFLOW_DEFINITION,
-				WorkContextDelegate.Resource.NAME, workflowName);
-		return WorkFlowEngineBuilder.aNewWorkFlowEngine().build().run(workFlow, workContext);
+		WorkFlowExecution workFlowExecution = saveWorkFlow(UUID.fromString(projectId), workFlowDefinition.getId(),
+				WorkFlowStatus.IN_PROGRESS, null, arguments);
+		WorkContextUtils.updateWorkContextPartially(workContext, projectId, workflowName, workFlowExecution.getId());
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java]: yes it looks better

**Review by RichardW98 — APPROVED** [review]

**Pull request merged**